File: src/main/java/net/yakclient/mixin/internal/methodadapter/core/CoreOverwritePatternMatcher.java
package net.yakclient.mixin.internal.methodadapter.core;

import net.yakclient.mixin.internal.bytecode.PriorityInstruction;
import net.yakclient.mixin.internal.instruction.core.CoreInstruction;
import org.objectweb.asm.*;

import java.util.Queue;

public class CoreOverwritePatternMatcher extends CoreMixinPatternMatcher {
    public CoreOverwritePatternMatcher(MethodVisitor visitor, CoreInstruction instructions) {
        super(visitor, instructions);
    }

    @Override
    public void visitCode() {
        super.visitCode();

        this.executeInsn();
    }

    @Override
    public void visitInsn() {
    }

    @Override
    public void visitVarInsn(int opcode, int var) {
    }

    @Override
    public void visitTypeInsn(int opcode, String type) {
    }

    @Override
    public void visitFieldInsn(int opcode, String owner, String name, String descriptor) {
    }

    @Override
    public void visitMethodInsn(int opcode, String owner, String name, String descriptor) {
    }

    @Override
    public void visitMethodInsn(int opcode, String owner, String name, String descriptor, boolean isInterface) {
    }

    @Override
    public void visitInvokeDynamicInsn(String name, String descriptor, Handle bootstrapMethodHandle, Object... bootstrapMethodArguments) {
    }

    @Override
    public void visitJumpInsn(int opcode, Label label) {
    }

    @Override
    public void visitLabel(Label label) {
    }

    @Override
    public void visitLdcInsn(Object value) {
    }

    @Override
    public void visitIincInsn(int var, int increment) {
    }

    @Override
    public void visitTableSwitchInsn(int min, int max, Label dflt, Label... labels) {
    }

    @Override
    public void visitLookupSwitchInsn(Label dflt, int[] keys, Label[] labels) {
    }

    @Override
    public void visitMultiANewArrayInsn(String descriptor, int numDimensions) {
    }

    @Override
    public AnnotationVisitor visitInsnAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
        return null;
    }

    @Override
    public void visitTryCatchBlock(Label start, Label end, Label handler, String type) {
    }

    @Override
    public AnnotationVisitor visitTryCatchAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
        return null;
    }

    @Override
    public void visitLocalVariable(String name, String descriptor, String signature, Label start, Label end, int index) {
    }

    @Override
    public AnnotationVisitor visitLocalVariableAnnotation(int typeRef, TypePath typePath, Label[] start, Label[] end, int[] index, String descriptor, boolean visible) {
        return null;
    }

    @Override
    public void visitLineNumber(int line, Label start) {
    }
}

File: base/src/main/kotlin/net/yakclient/mixin/base/internal/methodadapter/core/CoreOverwritePatternMatcher.kt
package net.yakclient.mixin.base.internal.methodadapter.core

import net.yakclient.mixin.base.internal.instruction.Instruction
import org.objectweb.asm.*

class CoreOverwritePatternMatcher(
    visitor: MethodVisitor,
    instructions: Instruction
) : CoreMixinPatternMatcher(visitor, instructions) {

    override fun visitCode() {
        super.visitCode()
        this.executeInsn()
    }

    override fun visitInsn() {}
    override fun visitVarInsn(opcode: Int, `var`: Int) {}
    override fun visitTypeInsn(opcode: Int, type: String) {}
    override fun visitFieldInsn(opcode: Int, owner: String, name: String, descriptor: String) {}
    override fun visitMethodInsn(opcode: Int, owner: String, name: String, descriptor: String) {}
    override fun visitMethodInsn(opcode: Int, owner: String, name: String, descriptor: String, isInterface: Boolean) {}
    override fun visitInvokeDynamicInsn(
        name: String,
        descriptor: String,
        bootstrapMethodHandle: Handle,
        vararg bootstrapMethodArguments: Any
    ) {
    }

    override fun visitJumpInsn(opcode: Int, label: Label) {}
    override fun visitLabel(label: Label) {}
    override fun visitLdcInsn(value: Any) {}
    override fun visitIincInsn(`var`: Int, increment: Int) {}
    override fun visitTableSwitchInsn(min: Int, max: Int, dflt: Label, vararg labels: Label) {}
    override fun visitLookupSwitchInsn(dflt: Label, keys: IntArray, labels: Array<Label>) {}
    override fun visitMultiANewArrayInsn(descriptor: String, numDimensions: Int) {}
    override fun visitInsnAnnotation(
        typeRef: Int,
        typePath: TypePath,
        descriptor: String,
        visible: Boolean
    ): AnnotationVisitor? {
        return null
    }

    override fun visitTryCatchBlock(start: Label, end: Label, handler: Label, type: String) {}
    override fun visitTryCatchAnnotation(
        typeRef: Int,
        typePath: TypePath,
        descriptor: String,
        visible: Boolean
    ): AnnotationVisitor? {
        return null
    }

    override fun visitLocalVariable(
        name: String,
        descriptor: String,
        signature: String,
        start: Label,
        end: Label,
        index: Int
    ) {
    }

    override fun visitLocalVariableAnnotation(
        typeRef: Int,
        typePath: TypePath,
        start: Array<Label>,
        end: Array<Label>,
        index: IntArray,
        descriptor: String,
        visible: Boolean
    ): AnnotationVisitor? {
        return null
    }

    override fun visitLineNumber(line: Int, start: Label) {}
}